wa/workloads/googlemaps/uiauto/app/src/main/java/com/arm/wa/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,149 @@
+                                                            .className("android.widget.TextView"));
+        if (!match.waitForExists(uiAutoTimeout)) {
+            throw new UiObjectNotFoundException("Could not find search result containing \"" + str + "\".");
+        }
+        match.click();
+    }
+
+    public void getDirectionsFromLocation() throws Exception {
+        UiObject directions = mDevice.findObject(new UiSelector().resourceId(packageID + "placepage_directions_button"));
+        directions.clickAndWaitForNewWindow(uiAutoTimeout);
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
+        // Search for and select Cambridge train station
+        search("Cambridge Station CBG, England", "search_omnibox_text_box");
+        selectSearchResultContaining("[CBG]");
+
+        // Dismiss the tutorial dialog for place information if it appears
+        UiObject gotItButton = mDevice.findObject(new UiSelector().resourceId(packageID + "tutorial_pull_up_got_it"));
+        if (gotItButton.waitForExists(uiAutoTimeout)) {
+            gotItButton.clickAndWaitForNewWindow(uiAutoTimeout);
+        }
+        sleep(3);
+
+        // Pinch to zoom, scroll around
+        UiObject mapContainer = mDevice.findObject(new UiSelector().resourceId(packageID + "mainmap_container"));
+        uiDeviceSwipeDown(100);
+        sleep(1);
+        uiDeviceSwipeUp(200);
+        sleep(1);
+        uiObjectVertPinchIn(mapContainer, 100, 25);
+        sleep(1);
+        uiDeviceSwipeLeft(100);
+        uiDeviceSwipeUp(100);
+        uiObjectVertPinchOut(mapContainer, 100, 50);
+        sleep(3);
+
+        // Get directions from Cambridge train station to Corpus Christi college
+        getDirectionsFromLocation();
+        search("Corpus Christi, Cambridge, England", "directions_startpoint_textbox");
+        selectSearchResultContaining("Corpus Christi College");
+        sleep(3);
+
+        // View the steps for the route
+        UiObject steps = mDevice.findObject(new UiSelector().textContains("STEPS & MORE")
+                                                            .className("android.widget.TextView"));
+        steps.clickAndWaitForNewWindow(uiAutoTimeout);
+
+        // Preview the first three steps of the route
+        UiObject preview, next;
+        preview = mDevice.findObject(new UiSelector().resourceId(packageID + "start_button"));
+        preview.clickAndWaitForNewWindow(uiAutoTimeout);
+        next = getUiObjectByDescription("Show next", "android.widget.ImageView");
+        for (int i = 0; i < 3; ++i) {
+            next.click();
+            sleep(1);
+        }
+
+        // Return to the normal map view
+        pressBack();
+        pressBack();
+        pressBack();
+    }
+
+    @Test
+    public void teardown() throws Exception {
+        unsetScreenOrientation();
+    }
+
+    public void runApplicationSetup() throws Exception {
+        // Dismiss a dialog regarding real-time traffic updates
```

Comment:
Might need to add another check here for a "Getting the most from Google" splash screen which can be shown as the first thing upon a new installation of google maps. I've attached a screenshot and UX dump.
[wk1-googlemaps-1-attempt01.tar.gz](https://github.com/ARM-software/workload-automation/files/2118681/wk1-googlemaps-1-attempt01.tar.gz)